Update on SwiftJavaDocumentation (#372)
swiftlang/swift-java: Sources/SwiftJavaDocumentation/Documentation.docc/index.md

### Getting started

> **SwiftJava** provides Java and Swift interoperability with minimal overhead. It eliminates the complex, error-prone process such as manually compiling Java classes to C headers for native access. With SwiftJava, developers can achieve a flexible, safe, and high-performance connection between the two languages.

Reviewer: This comes of a bit weird, no-one is manually compiling java classes in either approach. Maybe something like

```suggestion
**SwiftJava** provides a set of tools and libraries to enable Java and Swift interoperability. It allows developers to generate bindings to either language from the other, by using either source generation (for Java consuming Swift code), or by using a combination of Swift macros and source generation (for Swift consuming Java libraries).

The generated code is highly efficient, and less error-prone than manually writing the mappings by hand, and also guarantees memory safety across the boundaries between the languages.
```

> **Interoperability promotes**

Reviewer: ```suggestion
Reasons why you might want to reach for Swift and Java interoperability include, but are not limited to, the following scenarios:
```

> - Incremental adoption of Swift

Reviewer: ```suggestion
- Incremental adoption of Swift, in an existing Java codebase
```

- Reuse of libraries across languages

>     - Access libraries with or without native APIs

Reviewer: ```suggestion
    - Reuse existing libraries which exist in one ecosystem, but don't have a direct equivalent in the other
```

    - Implement performance-sensitive code in Swift
- Support for ecosystem-specific build tools

> **Tools**

Reviewer: ```suggestion
SwiftJava is offering a number of core libraries which support the language interoperability:
```
